moesif-servlet/src/main/java/com/moesif/servlet/BatchProcessor.java

```diff
@@ -0,0 +1,150 @@
+package com.moesif.servlet;
+
+import java.util.TimerTask;
+import java.util.logging.Logger;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.moesif.api.MoesifAPIClient;
+import com.moesif.api.http.client.APICallBack;
+import com.moesif.api.http.client.HttpContext;
+import com.moesif.api.http.response.HttpResponse;
+import com.moesif.api.models.EventModel;
+
+/***
+ * TimerTask to maintain the events queue and send it to moesif periodically.
+ *  Task runner method sends event in batches if at least one batch of events available. It
+ *  maintains :
+ *   1. queue to hold events and events related configuration settings.
+ *   2. flag to show if a job is in-progress/running.
+ *
+ *   It prints few helpful log statements, if debug has been enabled.
+ */
+public class BatchProcessor extends TimerTask {
+
+    private static final Logger logger = Logger.getLogger(BatchProcessor.class.toString());
+
+    private final MoesifConfiguration moesifConfig;
+    private final MoesifAPIClient moesifApi;
+    private final Boolean debug;
+    private final BlockingQueue<EventModel> batchQueue;    // Queue to hold the batched events
+
+    private boolean jobRunning = false;
+
+    public BatchProcessor(MoesifAPIClient moesifApi, MoesifConfiguration moesifConfig, boolean debug) {
+        this.moesifApi = moesifApi;
+        this.moesifConfig = moesifConfig;
+        this.debug = debug;
+
+        // Initialize the queue with queue size capacity.
+        this.batchQueue = new ArrayBlockingQueue<>(this.moesifConfig.eventQueueSize);
+    }
+
+    public int getBatchMaxTimeInSec() {
+        return this.moesifConfig.batchMaxTimeInSec;
+    }
+
+    public int getUpdateConfigTimeInMin() {
+        return this.moesifConfig.updateConfigTimeInMin;
+    }
+
+    /***
+     * Method to add event into a queue for batch-based event transfer. It maintains
+     * time when the last event was added to the event queue.
+     * @param event: EventModel
+     */
+    public void addEvent(EventModel event) {
+        try {
+            // Add the event to the queue
+            this.batchQueue.put(event);
+
+            if (this.debug) {
+                String msg = String.format("Event successfully added to queue. Queue Size = %d", this.batchQueue.size());
+                logger.info(msg);
+            }
+
+        } catch (InterruptedException e) {
+            logger.warning("Add event failed. " + e.toString());
+        } catch (Exception e) {
+            logger.warning("Add event failed. " + e.toString());
```

Comment:
minor nitpic: since we perform same action, do we need two catch here?

Reply:
Thanks @CodeYouMust for checking. No, we do not. I added it to keep IntelliJ happy (probably suggested based on their linting rule). Will remove it.